### ISSUE TITLE
Utf8Formatter.TryFormat(Double/Single) allocates a String.

### ISSUE THREAD

**Issue by AtsushiKan** [dotnet/corefx#25077]
As it's a thin wrapper around the classic BCL api. The algorithm is extremely complex and we don't have a C# version and porting the C++ from CoreCLR would be time-consuming and it's not clear whether the engineering result would be to our pleasing. For now, this overload is more for completeness than a cloud need so for now, this issue is for tracking.

**Comment by ahsonkhan** [comment]
> The algorithm is extremely complex and we don't have a C# version and porting the C++ from CoreCLR would be time-consuming and it's not clear whether the engineering result would be to our pleasing. For now, this overload is more for completeness than a cloud need so for now

Adjusting milestone to future.

**Comment by Tornhoof** [comment]
Would you accept a PR for an easy fix for this?
https://github.com/dotnet/corefx/blob/92a697cc5f49f756f3066d750db265835d5e66b4/src/Common/src/CoreLib/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Float.cs#L93-L94

and add a char stackalloc instead of the ToString  ( max double buffer length is 32, according to https://github.com/dotnet/coreclr/blob/9e7583d1402957472e41b6364112a69040399313/src/System.Private.CoreLib/shared/System/Number.Formatting.cs#L251)?
and use the UTF16 TryFormat api to format into the char span?

I did the same in SpanJson and the memory usage is obviously better.

@stephentoub @ahsonkhan

**Comment by stephentoub** [comment]
> Would you accept a PR for an easy fix for this?

Yes (in coreclr).  Previously we didn't do this because the implementation needed to build on top of netstandard, and the TryFormat API was netcoreapp-only.  Now that the implementation has moved into corelib and is also only built for netcoreapp, it's fine to utilize.
https://github.com/dotnet/coreclr/blob/9e7583d1402957472e41b6364112a69040399313/src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Float.cs#L94
